xla/tests/onednn_matmul_test.cc

```diff
@@ -1442,6 +1442,52 @@ TEST_F(MatmulTest, WeightsPrepackAndScratch) {
   )");
 }
 
+TEST_F(MatmulTest, ConsecutiveBinaryAdd) {
+  const char* matmul_module_str = R"(
+  HloModule matmul.test.f32
+    region_0.22 {
+    Arg_0.23 = f32[] parameter(0)
+    Arg_1.24 = f32[] parameter(1)
+  ROOT add.25 = f32[] add(Arg_0.23, Arg_1.24)
+  }
+
+  region_1.29 {
+    Arg_0.30 = f32[] parameter(0)
+    Arg_1.31 = f32[] parameter(1)
+    ROOT add.32 = f32[] add(Arg_0.30, Arg_1.31)
+  }
+
+  ENTRY main {
+    constant.2 = f32[] constant(1e-06)
+    broadcast.3 = f32[1000000] broadcast(constant.2), dimensions={}
+    constant.7 = f32[] constant(1)
+    broadcast.8 = f32[1000000,3] broadcast(constant.7), dimensions={}
+    Arg_0.1 = f32[3] parameter(0)
+    reshape.10 = f32[1,3] reshape(Arg_0.1)
+    broadcast.11 = f32[1,3] broadcast(reshape.10), dimensions={0,1}
+    reshape.12 = f32[3] reshape(broadcast.11)
+    broadcast.13 = f32[1000000,3] broadcast(reshape.12), dimensions={1}
+    subtract.14 = f32[1000000,3] subtract(broadcast.8, broadcast.13)
+    constant.4 = f32[] constant(0)
+    broadcast.5 = f32[3,3] broadcast(constant.4), dimensions={}
+    dot.15 = f32[1000000,3] dot(subtract.14, broadcast.5), lhs_contracting_dims={1}, rhs_contracting_dims={0}
```

Comment:
Can we reduce the ops to just necessary ops that reproduce the failure? I don't think all the dots are needed.

Reply:
The bug seems to be seen with this particular case.